src/rgw/rgw_data_sync.cc

```diff
@@ -3481,20 +3514,44 @@ class RGWListRemoteBucketCR: public RGWCoroutine {
   }
 };
 
+struct next_bilog_result {
+  uint64_t generation = 0;
+  int num_shards = 0;
+
+  void decode_json(JSONObj *obj) {
+    JSONDecoder::decode_json("generation", generation, obj);
+    JSONDecoder::decode_json("num_shards", num_shards, obj);
+  }
+};
+
+struct bilog_list_result {
+  list<rgw_bi_log_entry> entries;
+  bool truncated{false};
+  next_bilog_result next_log;
```

Comment:
i think this should be a `std::optional<next_bilog_result>` so we can tell whether or not it was present

there's an overload of `JSONDecoder::decode_json()` for `std::optional<T>` so the decode should work unchanged

Reply:
ack